File: python/caesar.py
import string
# alphabet = list(string.ascii_lowercase)
symbols = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890 !? ."

# First - Create new array with the rotated values:
def rot(key):
    newAlph = []
    for letter in symbols:
        index = (symbols.index(letter) + key) - 1
        # print(index)
        if(index >= len(symbols)):
            index = index - len(symbols)
            newAlph.append(symbols[index])
        else:
            newAlph.append(symbols[index])
    return newAlph

def encrypt(message):
    key = int(input(f"Enter key between 1-{len(symbols)}\n"))
    newAlphabet = rot(key)
    encrypted = []

    for char in message:
        letter = newAlphabet[symbols.index(char)]
        encrypted.append(letter)

    newMessage = ''.join(encrypted);

    print(newMessage)


message = input("Please enter a word to encrypt...\n")
encrypt(message)

File: javascript/caesar.js
const alphabet = ["a", "b", "c", "d","e", "f", "g", "h","i", "j", "k", "l","m", "n", "o", "p","q", "r", "s", "t","u", "v", "w", "x","y", "z"]

function rot(rot){
    const newAlph = []
    for(let i = 0; i < alphabet.length; i++){
        let index = (i + rot) - 1
        if(index >= 26){
            index = index - 26
            newAlph.push(alphabet[index])
        }else{
            newAlph.push(alphabet[index])
        }
    }
    return newAlph;
}

function encrypt(message, rotation){
    const newAlph = rot(rotation);
    const encrypted = [];

    message.split("").forEach(char => {
        encrypted.push(newAlph[alphabet.indexOf(char)])
    });
    return encrypted.join("")
}

encrypt('ezra', 5)